Add `Argilla` and `distilabel` to `https://huggingface.co/docs/hub/datasets-libraries` (#1340)
huggingface/hub-docs: docs/hub/datasets-argilla.md

> # Argilla

Reviewer: You might also want to update https://github.com/huggingface/hub-docs/blob/main/docs/hub/_toctree.yml

> Argilla is a collaboration tool for AI engineers and domain experts who need high-quality outputs, full data ownership, and overall efficiency in their work.

Reviewer 2: ```suggestion
Argilla is a collaboration tool for AI engineers and domain experts who need to build high quality datasets for their projects.
```

Reviewer 2: I know we have this in the readme but I think we can use a more direct and less salesy

![image](https://github.com/user-attachments/assets/0e6ce1d8-65ca-4211-b4ba-5182f88168a0)

> Whether you are working on monitoring and improving complex generative tasks involving LLM pipelines with RAG or you are working on a predictive task for things like AB-testing of span and text-classification models. Argilla helps you ensure your data work pays off. Argilla helps improve your AI output quality through data quality, take control of your data and models, and improve efficiency by quickly iterating on the right data and models.

Reviewer 2: ```suggestion
Argilla can be used for collecting human feedback for a wide variety of AI projects like traditional NLP (text classification, NER, etc.), LLMs (RAG, preference tuning, etc.), or multimodal models (text to image, etc.). Argilla's programmatic approach lets you build workflows for continuous evaluation and model improvement. The goal of Argilla is to ensure your data work pays off by quickly iterating on the right data and models.
```

> Whether you are working on monitoring and improving complex generative tasks involving LLM pipelines with RAG, or you are working on a predictive task for things like AB-testing of span- and text-classification models. Our versatile tool helps you ensure your data work pays off. We help you improve your AI output quality through data quality, take control of your data and models, and improve efficiency by quickly iterating on the right data and models.

Reviewer: Let's try to write in third person (no "we")

```suggestion
Whether you are working on monitoring and improving complex generative tasks involving LLM pipelines with RAG or you are working on a predictive task for things like AB-testing of span and text-classification models. Argilla helps you ensure your data work pays off. Argilla helps improve your AI output quality through data quality, take control of your data and models, and improve efficiency by quickly iterating on the right data and models.
```

## What do people build with Argilla?

> Our community uses Argilla to create amazing open-source [datasets](https://huggingface.co/datasets?library=library:argilla&sort=trending) and [models](https://huggingface.co/models?other=distilabel). 

Reviewer: ```suggestion
The community uses Argilla to create amazing open-source [datasets](https://huggingface.co/datasets?library=library:argilla&sort=trending) and [models](https://huggingface.co/models?other=distilabel). 
```

### Open-source datasets and models

**We love contributions to open-source** ourselves too.

> - Our [cleaned UltraFeedback dataset](https://huggingface.co/datasets/argilla/ultrafeedback-binarized-preferences-cleaned) and the [Notus](https://huggingface.co/argilla/notus-7b-v1) and [Notux](https://huggingface.co/argilla/notux-8x7b-v1) models, where we improved benchmark and empirical human judgment for the Mistral and Mixtral models with cleaner data using **human feedback**.

Reviewer 2: ```suggestion
- [Cleaned UltraFeedback dataset](https://huggingface.co/datasets/argilla/ultrafeedback-binarized-preferences-cleaned) and the [Notus](https://huggingface.co/argilla/notus-7b-v1) and [Notux](https://huggingface.co/argilla/notux-8x7b-v1) models, where we improved benchmark and empirical human judgment for the Mistral and Mixtral models with cleaner data using **human feedback**.
```

> - [distilabeled Intel Orca DPO dataset](https://huggingface.co/datasets/argilla/distilabel-intel-orca-dpo-pairs) and the [improved OpenHermes model](https://huggingface.co/argilla/distilabeled-OpenHermes-2.5-Mistral-7B), show how we improve model performance by filtering out 50% of the original dataset through **human and AI feedback**.

Reviewer 3: maybe avoid using "we" which doesn't refer to anyone in a doc page

> ### Internal Use cases

Reviewer 2: fully agree!

> - Our [distilabeled Intel Orca DPO dataset](https://huggingface.co/datasets/argilla/distilabel-intel-orca-dpo-pairs) and the [improved OpenHermes model](https://huggingface.co/argilla/distilabeled-OpenHermes-2.5-Mistral-7B), show how we improve model performance by filtering out 50% of the original dataset through **human and AI feedback**.

Reviewer 2: ```suggestion
- [distilabeled Intel Orca DPO dataset](https://huggingface.co/datasets/argilla/distilabel-intel-orca-dpo-pairs) and the [improved OpenHermes model](https://huggingface.co/argilla/distilabeled-OpenHermes-2.5-Mistral-7B), show how we improve model performance by filtering out 50% of the original dataset through **human and AI feedback**.
```

- AI for good: [the Red Cross presentation](https://youtu.be/ZsCqrAhzkFU?feature=shared) showcases **how their experts and AI team collaborate** by classifying and redirecting requests from refugees of the Ukrainian crisis to streamline the support processes of the Red Cross.
- Customer support: during [the Loris meetup](https://youtu.be/jWrtgf2w4VU?feature=shared) they showed how their AI team uses unsupervised and few-shot contrastive learning to help them **quickly validate and gain labelled samples for a huge amount of multi-label classifiers**.
- Research studies: [the showcase from Prolific](https://youtu.be/ePDlhIxnuAs?feature=shared) announced their integration with our platform. They use it to actively **distribute data collection projects** among their annotating workforce. This allows them to quickly and **efficiently collect high-quality data** for their research studies.

## Importing and exporting datasets and records

> This guide provides an overview of how to import and export your dataset or its records to the Hugging Face Hub.

Reviewer 2: ```suggestion
This guide shows how to import and export your dataset to the Hugging Face Hub.
```

In Argilla, you can import/export two main components of a dataset:
- The dataset's complete configuration defined in `rg.Settings`. This is useful if your want to share your feedback task or restore it later in Argilla.
- The records stored in the dataset, including `Metadata`, `Vectors`, `Suggestions`, and `Responses`. This is useful if you want to use your dataset's records outside of Argilla.

> 

Reviewer 2: This two blocks seem to be a bit disconnected (anyway I suggest removing the link and making it more clear that you need to deploy Argilla in Spaces. Maybe use this link instead: https://argilla-io.github.io/argilla/latest/getting_started/quickstart/#run-the-argilla-server

> 🚀 Try the Argilla 🤝 Hugging Face [deployment integration on Spaces](https://huggingface.co/docs/hub/en/spaces-sdks-docker-argilla)!

Reviewer 2: I'd remove this link and mention because we're going through a major overhaul and this guide it's always been kind of obsolete/difficult to maintain. Once we're done with the overhaul we can rewrite it to be super simple and easy to maintain.

# Argilla

> Argilla is a collaboration platform for AI engineers and domain experts that require high-quality outputs, full data ownership, and overall efficiency.

Reviewer 2: 💯  @davidberenstein1957 change to tool too.

## Importing and exporting datasets and records

This guide provides an overview of how to import and export your dataset or its records to the Hugging Face Hub.

In Argilla, you can import/export two main components of a dataset:
- The dataset's complete configuration defined in `rg.Settings`. This is useful if your want to share your feedback task or restore it later in Argilla.
- The records stored in the dataset, including `Metadata`, `Vectors`, `Suggestions`, and `Responses`. This is useful if you want to use your dataset's records outside of Argilla.

🚀 Try the Argilla 🤝 Hugging Face [deployment integration on Spaces](https://huggingface.co/docs/hub/en/spaces-sdks-docker-argilla)!

## Prerequisites

First [login with your Hugging Face account](../huggingface_hub/quick-start#login):

```bash
huggingface-cli login
```

Make sure you have `argilla>=2.0.0` installed:

```bash
pip install -U argilla
```

> ## Push an Argilla dataset to the Hugging Face Hub

Reviewer 3: maybe give some context that argilla's goal is to create datasets and so this is the step when the dataset is ready ?

feel free to add a screenshot as well :)

```

The `rg.Dataset.from_hub` method loads the configuration and records from the dataset repo. If you only want to load records, you can pass a `datasets.Dataset` object to the `rg.Dataset.log` method. This enables you to configure your own dataset and reuse existing Hub datasets. 

### With or without records

The example above will pull the dataset's `Settings` and records from the hub. If you only want to pull the dataset's configuration, you can set the `with_records` parameter to `False`. This is useful if you're just interested in a specific dataset template or you want to make changes in the dataset settings and/or records.

```python
dataset = rg.Dataset.from_hub(repo_id="<repo_id>", with_records=False)
```

With the dataset's configuration you could then make changes to the dataset. For example, you could adapt the dataset's settings for a different task:

```python
dataset.settings.questions = [rg.TextQuestion(name="answer")]
```

You could then log the dataset's records using the `load_dataset` method of the `datasets` package and pass the dataset to the `rg.Dataset.log` method.

```python
hf_dataset = load_dataset("<repo_id>")
dataset.log(hf_dataset)
```

> 

Reviewer 3: maybe add a section on what people can do after loading a dataset from the hub ? (edit/labelize/whatever)
